### ISSUE TITLE
Launching more tasks causes failure with retry messages.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#561]
The following script fails.

```python
@ray.remote
def f(x):
  return 1

def g(n):
  x = 1
  for i in range(n):
    x = f.remote(x)
  return x

ray.get([g(1000) for _ in range(100)])
```

It will indefinitely print a message like the following.

```
[WARN] (/Users/rkn/Workspace/ray/src/common/state/table.cc:85) retrying operation task_table_test_and_update, retry_count = -1
```

**Comment by robertnishihara** [comment]
I just tried this with 10 Redis shards, and the same problem occurs.

**Comment by robertnishihara** [comment]
I also tried removing the event log (in addition to 10 Redis shards). A local scheduler still gets marked as dead.

**Comment by robertnishihara** [comment]
When I say "removing the event log", I mean using the change in #572. Similarly, when I say "removing retries", I mean using the change in #574.

**Comment by ericl** [comment]
When this happens, the local_scheduler process is stuck at 100% cpu, and appears to be spending most of its time in `common/thirdparty/ae/ae.c:aeDeleteTimeEvent()`. It looks like each call to `aeDeleteTimeEvent()` scans the entire event queue, so assuming you eventually want to delete each event, the overall CPU used for event processing is going to grow O(n^2) wrt the size of the queue.

Because of this n^2 thing, I think that launching enough tasks is making event processing *slow enough* that the event queue grows unboundedly. Indeed it seems that once the queue size hits >6k events, it does start growing unboundedly and you get the retry messages.

TLDR: I think event processing is O(n^2) and can't keep up if there are too many tasks. Thoughts on angle of attack?

**Comment by robertnishihara** [comment]
Addressed by #649.
